src/mbgl/style/style.cpp

```diff
@@ -198,6 +198,16 @@ Layer* Style::getLayer(const std::string& id) const {
 Layer* Style::addLayer(std::unique_ptr<Layer> layer, optional<std::string> before) {
     // TODO: verify source
 
+    //Guard against duplicate layer ids
+    auto it = std::find_if(layers.begin(), layers.end(), [&](const auto& existing) {
+        return existing->getID() == layer->getID();
+    });
+
+    if (it != layers.end()) {
+        std::string msg = "Layer " + layer->getID() + " already exists";
+        throw std::runtime_error(msg.c_str());
```

Comment:
no need to to `c_str()` here; `std::runtime_error` also accepts `std::string`. You could change this to something like this;

```c++
throw std::runtime_error(std::string{"Layer "} + layer->getID() + " already exists");
```